### PR TITLE
Add db methods necessary for retrieving a random row

### PR DESCRIPTION
**Pull request by msgmaxim**
- plus better error handling in db code

### CODE REVIEW

File: storage/src/Database.cpp [hunk 1]
```diff
@@ -129,13 +130,101 @@ void Database::open_and_prepare(const std::string& db_path) {
...
+    item.ttl = sqlite3_column_int64(stmt, 2);
+    item.timestamp = sqlite3_column_int64(stmt, 3);
+    item.expiration_timestamp = sqlite3_column_int64(stmt, 4);
+    item.nonce = std::string((const char*)sqlite3_column_text(stmt, 5));
+    item.data = std::string((char*)sqlite3_column_blob(stmt, 6),
+                            sqlite3_column_bytes(stmt, 6));
+    return item;
+}
+
+bool Database::retrieve_by_index(uint64_t index, Item& item) {
```
Comment (sachaaaaa) [inline storage/src/Database.cpp]: What will be the purpose of this function?
(The index can't be relied upon to identify messages across snodes)

Comment (msgmaxim) [reply storage/src/Database.cpp]: If I know that there is 100 messages, I can generate a random number in [0; 100) and request it using this method.

Comment (sachaaaaa) [reply storage/src/Database.cpp]: My question was more related to the purpose of the function in regard to the  storage tests :)

Comment (msgmaxim) [reply storage/src/Database.cpp]: I will pick a random message and send its owner's pubkey to the node being tested. That node will need to find it in the database and return the message/hash.

Comment (sachaaaaa) [reply storage/src/Database.cpp]: Ah ok all good then !

**Pull request merged**